engine/src/witness/btc/btc_deposits.rs

```diff
@@ -65,15 +66,27 @@ impl<Inner: ChunkedByVault> ChunkedByVaultBuilder<Inner> {
 
 				// Submit all deposit witnesses for the block.
 				if !deposit_witnesses.is_empty() {
-					process_call(
-						pallet_cf_ingress_egress::Call::<_, BitcoinInstance>::process_deposits {
-							deposit_witnesses,
-							block_height: header.index,
-						}
-						.into(),
-						epoch.index,
-					)
-					.await;
+					if is_prewitness {
+						process_call(
```

Comment:
Can you factor out the process_call here, its basically impossible to see the difference etween these two branches otherwise.